pkg/pool-hooks/contracts/MevHook.sol

```diff
@@ -18,18 +19,25 @@ import {
     MAX_FEE_PERCENTAGE
 } from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
 
+import { EnumerableSet } from "@balancer-labs/v3-solidity-utils/contracts/openzeppelin/EnumerableSet.sol";
 import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+import { CastingHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/CastingHelpers.sol";
 
 import { SingletonAuthentication } from "@balancer-labs/v3-vault/contracts/SingletonAuthentication.sol";
 import { VaultGuard } from "@balancer-labs/v3-vault/contracts/VaultGuard.sol";
 import { BaseHooks } from "@balancer-labs/v3-vault/contracts/BaseHooks.sol";
+import { BalancerContractRegistry } from "@balancer-labs/v3-vault/contracts/BalancerContractRegistry.sol";
 
 contract MevHook is BaseHooks, SingletonAuthentication, VaultGuard, IMevHook {
+    using CastingHelpers for *;
     using FixedPoint for uint256;
+    using EnumerableSet for EnumerableSet.AddressSet;
 
     // Max Fee is 99.9999% (Max supported fee by the vault).
     uint256 private constant _MEV_MAX_FEE_PERCENTAGE = MAX_FEE_PERCENTAGE;
 
+    BalancerContractRegistry internal immutable _registry;
```

Comment:
```suggestion
    IBalancerContractRegistry internal immutable _registry;
```
Usually we use interfaces here. The only time we need the contract itself is if we're deploying it.